Parser/tokenizer.c

```diff
@@ -856,6 +859,25 @@ tok_nextc(struct tok_state *tok)
                 if (translated == NULL)
                     return EOF;
                 newtok = translated;
+                if (tok->stdin_content == NULL) {
+                    tok->stdin_content = PyMem_Malloc(strlen(translated) + 1);
+                    if (tok->stdin_content == NULL) {
+                        tok->done = E_NOMEM;
+                        return EOF;
+                    }
+                    strcpy(tok->stdin_content, translated);
+                    tok->stdin_content[strlen(translated)] = 0;
```

Comment:
You can use as well sprintf and avoid setting yourself the null character. Not sure if this can help merging a bit more both branches. In any case, feel free to ignore